semgrep/semgrep/test.py

```diff
@@ -159,18 +156,25 @@ def score_output_json(
             todo_ok_in_line = line_has_todo_ok(line)
             num_todo += int(todo_rule_in_line) + int(todo_ok_in_line)
 
-            if (not ignore_todo and todo_rule_in_line) or rule_in_line:
-                ruleid_lines[test_file_resolved][normalize_rule_id(line)].append(
-                    effective_line_num
-                )
-            if (not ignore_todo and todo_rule_in_line) or ok_in_line:
-                ok_lines[test_file_resolved][normalize_rule_id(line)].append(
-                    effective_line_num
-                )
-            if ignore_todo and todo_ok_in_line:
-                todo_ok_lines[test_file_resolved][normalize_rule_id(line)].append(
-                    effective_line_num
-                )
+            try:
+                if (not ignore_todo and todo_rule_in_line) or rule_in_line:
+                    rule_ids = normalize_rule_ids(line)
```

Comment:
We can pull the `rule_ids = normalize_rule_ids(line)` line out of the `if` statements since it's called the same way in each.

Reply:
Yeah, we could, but then it runs on every line rather than only the lines with comment syntaxes. It looks factorable, but I decided to leave it that way.